classes/migration/upgrade/v3_5_0/I9707_WeblateUILocales.php

```diff
@@ -14,19 +14,123 @@
+                        WHEN 'be@cyrillic' THEN 'be'
+                        WHEN 'bs' THEN 'bs_Latn'
+                        WHEN 'fr_FR' THEN 'fr'
+                        WHEN 'ja_JP' THEN 'ja'
+                        WHEN 'nb' THEN 'nb_NO'
+                        WHEN 'pt_PT' THEN 'pt'
+                        WHEN 'sr@cyrillic' THEN 'sr_Cyrl'
+                        WHEN 'sr@latin' THEN 'sr_Latn'
+                        WHEN 'uz@cyrillic' THEN 'uz'
+                        WHEN 'uz@latin' THEN 'uz_Latn'
+                        WHEN 'zh_CN' THEN 'zh_Hans'
+                        END"
+                    )])
+            );
+        }
+
+        // site supported and installed locales
+        $this->updateArrayLocale('site', 'supported_locales');
+        $this->updateArrayLocale('site', 'installed_locales');
+
+        // users locales
+        $this->updateArrayLocale('users', 'locales');
+
+        // context supported locales
+        $supportedDefaultSubmissionLocale = DB::table($this->getContextSettingsTable())
+            ->where('setting_name', '=', 'supportedDefaultSubmissionLocale')
+            ->get()
+            ->pluck('setting_value')
+            ->first();
+        if (in_array($supportedDefaultSubmissionLocale, array_keys($affectedLocales))) {
+            DB::table($this->getContextSettingsTable())
+                ->where('setting_name', '=', 'supportedDefaultSubmissionLocale')
+                ->update(['setting_value' => $affectedLocales[$supportedDefaultSubmissionLocale]]);
+        }
+
+        $contextLocaleSettingNames = [
+            'supportedFormLocales',
+            'supportedLocales',
+            'supportedSubmissionLocales',
+            'supportedAddedSubmissionLocales',
+            'supportedSubmissionMetadataLocales',
+        ];
+        $this->updateArrayLocale($this->getContextSettingsTable(), 'setting_value', $contextLocaleSettingNames);
+
+
+        // plugin_settings, customBlockManager
+        // assume that setting names blockTitle and blockContent are only used in customBlockManager
+        $blockLocalizedSettingNames = ['blockTitle', 'blockContent'];
+        $this->updateArrayLocale('plugin_settings', 'setting_value', $blockLocalizedSettingNames, true);
+    }
+
+    /**
+     * Update locale arrays, and locales as array keys
+     */
+    public function updateArrayLocale(string $tableName, string $columnName, ?array $settingNames = null, ?bool $key = false)
+    {
+        $affectedLocales = $this->getAffectedLocales();
+        array_shift($affectedLocales);
+        $replaceString = 'REPLACE(' . $columnName . ', \'"be@cyrillic"' . ($key ? ':' : '') . '\', \'"be"' . ($key ? ':' : '') . '\')';
+        foreach ($affectedLocales as $oldLocale => $newLocale) {
+            $replaceString = 'REPLACE(' . $replaceString . ', \'"' . $oldLocale . '"' . ($key ? ':' : '') . '\', \'"' . $newLocale . '"' . ($key ? ':' : '') . '\')';
+        }
```

Comment:
This way you can drop the fixed string:

```suggestion
$key = $key ? ':' : '';
$replaceString = $columnName;
foreach ($affectedLocales as $oldLocale => $newLocale) {
    $replaceString = "REPLACE({$replaceString}, '\"{$oldLocale}\"{$key}', '\"{$newLocale}\"{$key}')";
}
```